core/src/main/java/org/mskcc/cbio/portal/scripts/NormalizeExpressionLevels.java

```diff
@@ -283,259 +300,248 @@ private static void computeZScoreXP(String file){
                     }
-                    for(int k =0;k<SAMPLES;k++)
-                       outputLine.add( NOT_AVAILABLE );
-                    out.println( join( outputLine, "\t") );
+                    for (int k =0;k<SAMPLES;k++) {
+                        outputLine.add( NOT_AVAILABLE );
+                    }
+                    zscoreFileWriter.println( join( outputLine, "\t") );
                 }
                 genes++;
-            }else{
-               if (firstSamplePosition==1) {
+            } else {
+                if (firstSamplePosition==1) {
                     outputLine.add(values[0]);
                 } else {
                     outputLine.add(gene.getHugoGeneSymbolAllCaps());
                     outputLine.add(Long.toString(gene.getEntrezGeneId()));
                 }
-               for(int k =0;k<SAMPLES;k++)
-                  outputLine.add( NOT_AVAILABLE );
-               out.println( join( outputLine, "\t") );
+                for (int k =0;k<SAMPLES;k++) {
+                    outputLine.add( NOT_AVAILABLE );
+                }
+                zscoreFileWriter.println( join( outputLine, "\t") );
             }
-         }
-         if( 0 == genesFound ){
+        }
+        if ( 0 == genesFound ) {
             fatalError( "none of the genes in the expression file '" + file + "' were in the copy number file." );
-         }
-         if( 0 == rowsWithSomeDiploidCases ){
+        }
+        if ( 0 == rowsWithSomeDiploidCases ) {
             fatalError( "no genes with at least " + MIN_NUM_DIPLOIDS + " diploid cases found. Check copy number file." );
-         }
-      } catch (IOException e) {
-         fatalError( "cannot read from <expression_file> in '" + file + "'.");
-      }
-      out.close();
-   }
+        }
+        } catch (IOException e) {
+            fatalError( "cannot read from <expression_file> in '" + file + "'.");
+        }
+        zscoreFileWriter.close();
+    }
    
-   static int NO_POSITION = -1;
-   private static int getFirstDataColumn( String[] values){
-      // TODO: instead of guessing, we should normalizing
-         for(int i=0;i<values.length;i++) {
-             if (!values[i].equalsIgnoreCase("GENE SYMBOL")
-                     && !values[i].equalsIgnoreCase("SYMBOL")
-                     && !values[i].equalsIgnoreCase("HUGO_SYMBOL")
-                     && !values[i].equalsIgnoreCase("ENTREZ_GENE_ID")
-                     && !values[i].equalsIgnoreCase("LOCUS ID")
-                     && !values[i].equalsIgnoreCase("CYTOBAND")
-                     && !values[i].equalsIgnoreCase("LOCUS")
-                     && !values[i].equalsIgnoreCase("ID")
-                     && !values[i].equalsIgnoreCase("Composite.Element.REF")) {
-                 return i;
-             }
-         }
-            
-      
-      return NO_POSITION;
-      
-   }
+   private static final int NO_POSITION = -1;
```

Comment:
Since this is used in another location, I suggest moving this to the top with the other definitions.

Reply:
and using it in place of -1 on lines like 160, 165